store/v2/root/test_util.go

```diff
@@ -0,0 +1,78 @@
+package root
+
+import (
+	"bytes"
+	"errors"
+
+	dbm "github.com/cosmos/cosmos-sdk/db"
+)
+
+type dbDeleteVersionFails struct{ dbm.DBConnection }
+type dbRWCommitFails struct{ dbm.DBConnection }
+type dbRWCrudFails struct{ dbm.DBConnection }
+type dbSaveVersionFails struct{ dbm.DBConnection }
+type dbRevertFails struct {
+	dbm.DBConnection
+	// order of calls to fail on (eg. [1, 0] => first call fails; second succeeds)
+	failOn []bool
+}
+type dbVersionsIs struct {
+	dbm.DBConnection
+	vset dbm.VersionSet
+}
+type dbVersionsFails struct{ dbm.DBConnection }
+type rwCommitFails struct{ dbm.DBReadWriter }
+type rwCrudFails struct {
+	dbm.DBReadWriter
+	onKey []byte
+}
+
+func (dbVersionsFails) Versions() (dbm.VersionSet, error) { return nil, errors.New("dbVersionsFails") }
```

Comment:
Why no receiver? (Just a question - I've never seen a go method without a receiver)

Reply:
This is an interface wrapper which just exists to trigger a failure, so it doesn't actually need to use the receiver object